### PR TITLE
[RDY] Adds option to remove destroyed rooms for a fee

### LINKED ISSUE #969: Game error and glitch: doctor sprite stuck between doors
There was a sudden error and then I noticed a doctor inside a room being behind a door and in front of another (which should not be there), and the sprite is frozen.
[Screenshot here](https://i.imgur.com/uCY0YB4.jpg)

[gamelog.txt](https://github.com/CorsixTH/CorsixTH/files/109006/gamelog.txt)
[stuckdoctor.zip](https://github.com/CorsixTH/CorsixTH/files/109009/stuckdoctor.zip)

### LINKED ISSUE #991: Cleanup/remove exploded rooms
What is it you would like in the game?
The ability to delete broken rooms. Ideally this should be a non-default game setting.

What version of the product are you using? On what operating system?
Windows 10, latest production release of CorsixTH installed ontop of the GOG version.

### PR DESCRIPTION
**Pull request by mugmuggy**
*Fixes #991*
*Fixes #969*

**Describe what the proposed change does**
- Adds option to customize dialog to turn on the option to remove destroyed rooms
- Costs twice as much minus some 'scrap' recovery to remove the room.
- Destroyed rooms now reduce hospital value appropriately

### CODE REVIEW

File: CorsixTH/Lua/game_ui.lua [hunk 1]
```diff
@@ -643,10 +643,82 @@ function GameUI:onMouseUp(code, x, y)
...
-          self:setCursor(self.waiting_cursor)
-          self.edit_room = room
-          room:tryToEdit()
+        if room then
+          if not room.crashed then
+            self:setCursor(self.waiting_cursor)
+            self.edit_room = room
+            room:tryToEdit()
+          else
+            if self.app.config.remove_destroyed_rooms then
```
Comment (Alberth289346) [inline CorsixTH/Lua/game_ui.lua:652]: Fold this new code into a function, probably even more than one. Having to find a matching 'end' 80 lines down is a mess!

**Comment by Alberth289346** [comment]
I would consider this to be a cheat, at least wrt original behavior.

Maybe we should reconsider how we want to treat behavior extensions in general? Currently we just add it all in the configuration file, but maybe not all options are the same "kind"?

File: CorsixTH/Lua/game_ui.lua (hunk 1 above)
Comment (mugmuggy) [reply CorsixTH/Lua/game_ui.lua:652]: Indeed, it was just a slightly massaged guts of UIEditRoom:returnToDoorPhase, so can probably massage another method there to cover both scenarios.

**Comment by mugmuggy** [comment]
I considered it the same as build on pause, something that varied from the original giving the player some sort of an advantage, a plain configuration item.

By using twice the room cost, for the cheaper rooms, its nearly a break even opportunity for the most part to just expand to another small plot, more expensive rooms its more certainly a given to just expand.

The other advantage it gives by freeing up that space is the value can go up again, with this change subtracting away the crashed room value - which it should have done anyway. That gain is generally neutralised if the same room is built compared to not subtracting the value like occurs now.

Due to research reducing machine cost, the more researched a room is between building it and removing it further down the track, the buying another plot scenario diminshes, with the caveat that you don't want to expand anyway. Though if you have two or more crashed rooms, expansion is probably more cost effective in the short term.

**Comment by mugmuggy** [comment]
I'll squash this before merge if there are no other changes required

**Comment by TheCycoONE** [comment]
LGTM, I think some players will like this a lot.

**Comment by lukejames1111** [comment]
In the dialogue box I wish it told you what room you are about to destroy. After a massive earthquake I forget what rooms need rebuilding. If it said "Would you like to remove X-Ray for..." it would make it easier for me to know what to rebuild.

**Comment by lewri** [comment]
> In the dialogue box I wish it told you what room you are about to destroy. After a massive earthquake I forget what rooms need rebuilding. If it said "Would you like to remove X-Ray for..." it would make it easier for me to know what to rebuild.

No need to necro a merged PR. :) Make a feature request in the issues section and it can be addressed on its own merits then

**Pull request merged**